docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/migrate-to-an-out-of-tree-cloud-provider/migrate-to-out-of-tree-azure.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Migrating Azure In-tree to Out-of-tree
+---
+
+<head>
+  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/migrate-to-an-out-of-tree-cloud-provider/migrate-to-out-of-tree-azure"/>
+</head>
+
+Kubernetes is moving away from maintaining cloud providers in-tree. In Kubernetes v1.30 and later, the in-tree cloud providers have been removed. Rancher v2.9.0 allows you to upgrade to Kubernetes v1.30 when you migrate from an in-tree to out-of-tree provider.
```

Comment:
@kinarashah , suggesting to add note about 1.29 as well - my understanding is in-tree cloud provider won't work on 1.29 unless some K8s feature flags are updated (https://www.mirantis.com/blog/what-s-new-in-kubernetes-1-29-storage-improvements-windows-features-externalized-cloud-provider-integrations) so we should probably call it out and recommend upgrading on 1.29 as well.

Reply:
Added a note in the migration section too, 

> Starting with Kubernetes 1.29, in-tree cloud providers have been disabled. You must disable `DisableCloudProviders` and `DisableKubeletCloudCredentialProvider` to use the in-tree Azure cloud provider or migrate from in-tree cloud provider to out-of-tree provider. You can disable the required feature gates by setting `feature-gates=DisableCloudProviders=false` as an additional argument for the cluster's Kubelet, Controller Manager, and API Server in the advanced cluster configuration. Additionally, set `DisableKubeletCloudCredentialProvider=false` in the Kubelet's arguments to enable in-tree functionality for authenticating to Azure container registries for image pull credentials. See [upstream docs](https://github.com/kubernetes/kubernetes/pull/117503) for more details.